tests/functional/adapter/basic/test_data_types.py

```diff
@@ -12,9 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import pytest
 from dbt.tests.adapter.utils.data_types.test_type_boolean import BaseTypeBoolean
 from tests.fixtures.profiles import unique_schema, dbt_profile_data
+from tests.utils.util import (
+    check_relations_equal,
+    get_relation_columns
+)
+from dbt.tests.util import (
+    run_dbt
+)
 
 
+models__actual_sql = """
+select cast('True' as {{ type_boolean() }}) as boolean_col
+"""
+
 class TestTypeBoolean(BaseTypeBoolean):
-    pass
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"actual_view.sql": self.interpolate_macro_namespace(models__actual_sql, "type_boolean")}
+
+    def test_check_types_assert_match(self, project):
+        run_dbt(["build"])
+
+        # check contents equal
+        check_relations_equal(project.adapter, ["expected", "actual_view"])
+
+        # check types equal
+        expected_cols = get_relation_columns(project.adapter, "expected")
+        actual_cols = get_relation_columns(project.adapter, "actual_view")
+        print(f"Expected columns: {expected_cols}")
+        print(f"Actual columns: {actual_cols}")
+        self.assert_columns_equal(project, expected_cols, actual_cols)
```

Comment:
This is just overwriting the same function from the dbt testing code to use our custom `get_relation_columns` implementation instead of the dbt one.